app/channels/layer_channel.rb

```diff
@@ -20,17 +54,38 @@ def unsubscribed
 
   private
 
-  def persist_diff(data)
-    client_diff = data['diff']
-    return false if client_diff.nil?
-
-    seq = data['seq']
-    return false if seq.nil?
+  def persist_diff(diff)
+    diff_type = diff['diff_type']
+    diff_seq = diff['seq']
+    diff_visible = diff['visible']
+    diff_data = diff['data']
+    return false if diff_type.nil? || diff_seq.nil? || diff_data.nil?
 
     diff = Diff.new
-    diff.data = client_diff
-    diff.seq = seq
+    diff.diff_type = diff_type
+    diff.seq = diff_seq
+    diff.data = diff_data
+    diff.visible = diff_visible unless diff_visible.nil?
```

Comment:
should this check for `diff_visible.nil?` be included in the L62 checks?

Reply:
No since only tangible diffs (text and pen) will have values for the `visible` field. Other types of diffs will just have null values for the field.

(a bit of context)
`tangible` diffs are all we care about when we fetch existing diffs for a page. The other types of diffs apply side affects to the diffs in the database as they come in:
- `remove` diffs mark currently visible tangible diff records as invisible
- `translate` diffs replace the data field of existing tangible diff records with the translated data 
- `fetch-existing` essentially acts as a signal that an out of sync layer transmits to the server in order to refetch the existing visible tangible diffs for a layer (in order to refresh itself and get back in sync)
    - its a bit of a misnomer since this its more of a signal and is not actually stored in the database as a diff but gets the job done